python_bindings/src/PyDerivative.cpp

```diff
@@ -6,6 +6,7 @@ namespace PythonBindings {
 void define_derivative(py::module &m) {
     auto derivative_class = py::class_<Derivative>(m, "Derivative")
         .def("__getitem__", [](const Derivative &d, const Func &func) {return d(func);})
+        .def("__getitem__", [](const Derivative &d, const Func &func, int update_id) {return d(func, update_id);})
```

Comment:
This doesn't offer the same default value options as the C++ method does. You can do this easily in PyBind by doing something like

        .def("__getitem__", [](const Derivative &d, const Func &func, int update_id, bool bounded) {
        	return d(func, update_id, bounded);
        }, py::arg("func"), py::arg("update_id") = -1, py::arg("bounded") = true)

Reply:
(This would allow you to delete the previous single-arg version)

Reply:
Turns out that __getitem__ can only have a single argument, so this didn't work.
I used std::tuple instead.